superdesk/io/feed_parsers/ninjs.py

```diff
@@ -182,16 +186,31 @@ def parse_renditions(self, renditions):
                 rend[rendition_name] = parsed_rendition
         return rend
 
-    def _format_qcodes(self, items):
+    def _format_qcodes(
+        self, items: List[Dict[str, Any]], language: Optional[str] = None, cv_name: Optional[str] = None
+    ) -> List[Dict[str, Any]]:
         subjects = []
         for item in items:
-            subject = {"name": item.get("name"), "qcode": item.get("code")}
+            if cv_name:
```

Comment:
It would be better to retrieve the CV items before this for loop, and store the items there to use.
i.e.
```python
cv = get_resource_service("vocabularies").get_from_mongo(req=None, lookup={"_id": cv_name}) or {}
cv_items = {
    item["qcode"]: item
    for item in cv.get("items")
}
```

Then inside the for loop, you can use the following:
```python
if cv_item.get(item.get("code")):
    subject = cv_item[item["code"]
else:
    subject = {"name": item.get("name"), "qcode": item.get("code")}

if not subject.get("translations") and item.get("translations"):
    subject["translations"] = item["translations"]
if not subject.get("scheme") and item.get("scheme"):
    subject["scheme"] = item["scheme"]

subjects.append(subject)
```